File: src/main/java/com/test/java/util/ParamUtils.java
package com.test.java.util;

import com.alibaba.fastjson.JSON;

import java.io.*;
import java.util.Map;

/**
 * @author lvxw
 */
public class ParamUtils{
    /**
     * 读取classpath json格式的配置文件，并转成对象
     * @param path
     * @return
     */
    public static Map<String,Object> getClassPathFileContent(String path){
        ClassLoader classLoader = ParamUtils.class.getClassLoader();
        BufferedReader br = new BufferedReader(new InputStreamReader(classLoader.getResourceAsStream(path)));

        StringBuilder sb = new StringBuilder();
        String lineStr;
        try {
            while((lineStr = br.readLine()) != null) {
                sb.append(lineStr);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        return  jsonStrToMap(sb.toString());
    }

    /**
     * json字符串转map
     * @param jsonStr
     * @return
     */
    public static Map<String,Object> jsonStrToMap(String jsonStr){
        Map<String, Object> jsonMap = (Map<String, Object>) JSON.parse(jsonStr);
        return jsonMap;
    }
}

File: src/main/scala/com/test/scala/util/ParamUtils.scala
package com.test.scala.util

import scala.io.Source
import scala.util.parsing.json.JSON

object ParamUtils{

  /**
    * 读取classpath json格式的配置文件，并转成对象
    * @param path
    * @return
    */
  def getClassPathFileContent(path:String):Map[String,Any] ={
    val content = Source.fromURL(this.getClass.getClassLoader.getResource(path)).mkString
    jsonStrToMap(content).asInstanceOf[Map[String,Any]]
  }

  /**
    * json字符串转map
    * @param args
    * @return
    */
  def jsonStrToMap(jsonStr:String):Any ={
    val jsonObj = JSON.parseFull(jsonStr)
    jsonObj match {
      case Some(map: Map[String, String]) => map
      case _ => None
    }
  }
}
